File: todo-api/TodoApi/TodoApi/Models/Item.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace TodoApi.Models
{
    [BsonIgnoreExtraElements]
    public class Item
    {        
        public int Id { get; set; }

        public string Name { get; set; }

        public bool isChecked { get; set; }
    }
}

File: todo-app/src/modules/Item.js
export default class Item {
  constructor(id, name, isChecked) {
    this.id = id;
    this.name = name;
    this.isChecked = isChecked;
  }

  updateIsChecked(ischecked) {
    this.isChecked = ischecked;
  }

  static fromJson(json) {
    const item = new Item(json.id, json.name, json.ischecked);
    return item;
  }
}
